clang/lib/Sema/SemaExpr.cpp

```diff
@@ -215,17 +215,22 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
   if (getLangOpts().SYCLIsDevice) {
     if (auto VD = dyn_cast<VarDecl>(D)) {
       bool IsConst = VD->getType().isConstant(Context);
-      if (!IsConst && VD->getStorageClass() == SC_Static)
+      bool IsRuntimeEvaluated =
+          ExprEvalContexts.empty() ||
+          (!ExprEvalContexts.back().isUnevaluated() &&
+           !ExprEvalContexts.back().isConstantEvaluated());
```

Comment:
Should we be checking `Sema::isConstantEvaluated()` instead as that handles `isConstantEvaluatedOverride`? (Not certain what that thing is, however.)

Reply:
Ah, neat!  yeah, I think I there is a useful version of `isUnevaluated` (`Sema::isUnevaluatedContext`) that replaces the other condition as well.

It looks like we still have to check that the eval-context is `!empty`, but I have no idea when that is the case.